catalog/task/create-internal-request/0.7/create-internal-request.yaml

```diff
@@ -156,12 +156,11 @@ spec:
 
           IR="ir-$(params.pipelineRunName)-$(params.request)"
           while true; do
-            STATUS=\$(kubectl get internalrequest \${IR} -o jsonpath='{.status.conditions[0].status}')
-            case "\${STATUS}" in
-              True | False )
+            REASON=\$(kubectl get internalrequest \${IR} -o jsonpath='{.status.conditions[0].reason}')
```

Comment:
after further thought, I am wondering if we should alter the jsonpath to include the type:

(not sure of syntax)

`status.condition[].type = 'Succeeded'`

and not rely on ordering

Reply:
I think it adds some confusion. I am looking into a failed internalrequest and it is how it looks:
```yaml
      message: 'Tasks Completed: 1 (Failed: 1, Cancelled 0), Skipped: 0'
      reason: Failed
      status: "False"
      type: Succeeded
    startTime: "2023-08-30T16:46:49Z"
```
So, the reason is Failed but the type is Succeeded. If I understand the logic, it means that the internalrequest succeeded but the result was a failure.

Reply:
changed it to use `.status.conditions[]` instead of `.status.conditions[0]`

Reply:
> I think it adds some confusion. I am looking into a failed internalrequest and this is how it looks:
> 
> ```yaml
>       message: 'Tasks Completed: 1 (Failed: 1, Cancelled 0), Skipped: 0'
>       reason: Failed
>       status: "False"
>       type: Succeeded
>     startTime: "2023-08-30T16:46:49Z"
> ```
> 
> So, the reason is Failed but the type is Succeeded. If I understand the logic, it means that the internalrequest succeeded but the result was a failure.

https://github.com/redhat-appstudio/internal-services/blob/main/api/v1alpha1/internalrequest_types.go#L90 it basically means that the execution of it is complete and the result was failure

Reply:
I think that is what Scott was getting at. Instead of assuming the `Succeeded` type condition is `[0]`, use logic to specifically extract the `Succeeded` condition

Reply:
Ah, you mean that if we had some retry or reconcile for internalrequests we would have multiple conditions here. Got it now. I will do it.

Reply:
please take a look @scoheb @johnbieren